pkg/cloudprovider/aws/instance.go

```diff
@@ -43,6 +45,17 @@ type InstanceProvider struct {
 	launchTemplateProvider *LaunchTemplateProvider
 }
 
+func NewInstanceProvider(ec2api ec2iface.EC2API, instanceTypeProvider *InstanceTypeProvider, subnetProvider *SubnetProvider, ssm *ssm.SSM,
```

Comment:
I left this in since i had already created it - lmk if you'd rather me revert this.

Reply:
Either way. I hate constructors that span many lines -- in a sense, explicit object creation in go is very similar to the builder pattern.